cloud-agent/service/server/src/main/scala/org/hyperledger/identus/agent/server/jobs/IssueBackgroundJobs.scala

```diff
@@ -2,40 +2,68 @@ package org.hyperledger.identus.agent.server.jobs
 import org.hyperledger.identus.pollux.core.model.*
 import org.hyperledger.identus.pollux.core.model.error.CredentialServiceError
 import org.hyperledger.identus.pollux.core.service.CredentialService
+import org.hyperledger.identus.resolvers.DIDResolver
+import org.hyperledger.identus.shared.messaging
+import org.hyperledger.identus.shared.messaging.{Message, WalletIdAndRecordId}
+import org.hyperledger.identus.shared.messaging.MessagingService.RetryStep
+import org.hyperledger.identus.shared.models.{WalletAccessContext, WalletId}
 import org.hyperledger.identus.shared.models.Failure
 import org.hyperledger.identus.shared.utils.aspects.CustomMetricsAspect
 import org.hyperledger.identus.shared.utils.DurationOps.toMetricsSeconds
 import zio.*
 import zio.metrics.*
 
+import java.util.UUID
+
 object IssueBackgroundJobs extends BackgroundJobsHelper {
 
-  val issueCredentialDidCommExchanges = {
-    for {
+  private val TOPIC_NAME = "issue"
+
+  val issueFlowsHandler = messaging.MessagingService.consumeWithRetryStrategy(
+    "identus-cloud-agent",
+    IssueBackgroundJobs.handleMessage,
+    Seq(
+      RetryStep(TOPIC_NAME, 5, 0.seconds, s"$TOPIC_NAME-retry-1"),
+      RetryStep(s"$TOPIC_NAME-retry-1", 5, 2.seconds, s"$TOPIC_NAME-retry-2"),
+      RetryStep(s"$TOPIC_NAME-retry-2", 5, 4.seconds, s"$TOPIC_NAME-retry-3"),
+      RetryStep(s"$TOPIC_NAME-retry-3", 5, 8.seconds, s"$TOPIC_NAME-retry-4"),
+      RetryStep(s"$TOPIC_NAME-retry-4", 5, 16.seconds, s"$TOPIC_NAME-DLQ")
+    )
+  )
+
+  private def handleMessage(message: Message[UUID, WalletIdAndRecordId]): RIO[
+    HttpClient & DidOps & DIDResolver & (CredentialService & DIDNonSecretStorage & (ManagedDIDService & AppConfig)),
+    Unit
+  ] = {
+    (for {
+      _ <- ZIO.logInfo(s"!!! Handling recordId: ${message.value} via Kafka queue")
       credentialService <- ZIO.service[CredentialService]
-      config <- ZIO.service[AppConfig]
-      records <- credentialService
-        .getIssueCredentialRecordsByStatesForAllWallets(
-          ignoreWithZeroRetries = true,
-          limit = config.pollux.issueBgJobRecordsLimit,
-          IssueCredentialRecord.ProtocolState.OfferPending,
-          IssueCredentialRecord.ProtocolState.RequestPending,
-          IssueCredentialRecord.ProtocolState.RequestGenerated,
-          IssueCredentialRecord.ProtocolState.RequestReceived,
-          IssueCredentialRecord.ProtocolState.CredentialPending,
-          IssueCredentialRecord.ProtocolState.CredentialGenerated
-        )
-      _ <- ZIO
-        .foreachPar(records)(performIssueCredentialExchange)
-        .withParallelism(config.pollux.issueBgJobProcessingParallelism)
-    } yield ()
+      walletAccessContext = WalletAccessContext(WalletId.fromUUID(message.value.walletId))
+      record <- credentialService
+        .findById(DidCommID(message.value.recordId.toString))
+        .provideSome(ZLayer.succeed(walletAccessContext))
+        .someOrElseZIO(ZIO.dieMessage("Record Not Found"))
```

Comment:
I think the meta-information about the problematic record should be added to the dieMessage. Otherwise, troubleshooting the problems will be difficult.

Reply:
Done